rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/ExpandedGetResponseFilter.java

```diff
@@ -209,6 +216,17 @@ private void attachExpandedGetsBody(
 
       Optional<UriRouteMatch<Object, Object>> routeMatch = router.GET(targetUri);
 
+      for (Verb verb : contract.getVerbs()) {
+        for (Header header : verb.getHeaders()) {
+          String headerName = header.getName();
+
+          if (variables.get(headerName) == null) {
+            String variableName = LOWER_HYPHEN.to(LOWER_CAMEL, headerName.toLowerCase());
+            variables.put(variableName, extractHeader(request, header));
+          }
+        }
+      }
+
```

Comment:
There are some issues with this code section.

1. I would execute the whole for loop only in case that data is really needed. Means only if the routeMatch is present.
2. You iterate to each and every verb but that is wrong and error prone. An expands all requests all entities of a sub resource. Therefore the `COLLECTION_GET` word is your target here.
3. `extractHeader` can return `null` values. Do these have to be in the map?